### PR TITLE
Stealable queue enum

### PR DESCRIPTION
**Pull request by stellar-aria**
Converts StealableQueue to a strongly-type enum class

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,3 +29,9 @@ __pycache__
 .*.swp
 .*.swo
 .session.vim
+
+.idea
+*.srctrl*
+*.srctrl*
+*.srctrl*
```
Comment (m-m-adams) [inline .gitignore:36]: repeated

File: src/definitions_cxx.hpp [hunk 2]
```diff
@@ -907,20 +907,21 @@ enum class LoopType {
...
-	                                                     // to load it all again
-	NUM_STEALABLE_QUEUES,
-};
+enum class StealableQueue {
+	NO_SONG_SAMPLE_DATA,
+	NO_SONG_SAMPLE_DATA_CONVERTED, // E.g. from floating point file, or wrong endianness AIFF file.
+	NO_SONG_WAVETABLE_BAND_DATA,
+	NO_SONG_SAMPLE_DATA_REPITCHED_CACHE,
+	NO_SONG_SAMPLE_DATA_PERC_CACHE,
+	NO_SONG_AUDIO_FILE_OBJECTS,
```
Comment (m-m-adams) [inline src/definitions_cxx.hpp:916]: we should put a "TODO: having these in the stealable region is a bad idea in general" on this one

File: src/deluge/memory/cache_manager.cpp [hunk 3]
```diff
@@ -104,7 +107,7 @@ uint32_t CacheManager::ReclaimMemory(MemoryRegion& region, int32_t totalSizeNeed
 			}
 
 			// Ok, we've got one Stealable
-			uint32_t* __restrict__ header = (uint32_t*)((uint32_t)stealable - 4);
+			auto* __restrict__ header = std::bit_cast<uint32_t*>((uint32_t)stealable - 4);
```
Comment (m-m-adams) [inline src/deluge/memory/cache_manager.cpp:110]: If we're changing this anyway can we make it uintptr_t? No worries if that causes follow on issues

File: src/definitions_cxx.hpp (hunk 2 above)
Comment (stellar-aria) [reply src/definitions_cxx.hpp:917]: Done

File: .gitignore (hunk 1 above)
Comment (stellar-aria) [reply .gitignore]: Fixed

File: src/deluge/memory/cache_manager.cpp (hunk 3 above)
Comment (stellar-aria) [reply src/deluge/memory/cache_manager.cpp]: Fixed

**Review by m-m-adams — APPROVED** [review]

File: tests/unit/CMakeLists.txt [hunk 4]
```diff
@@ -21,33 +14,32 @@ add_compile_definitions(
...
+    # Used by most other modules
+    ../../src/deluge/util/*
 
+    # Mock implementations
+    mocks/*
+)
 
 add_executable(RunAllTests RunAllTests.cpp memory_tests.cpp)
-target_sources(RunAllTests PUBLIC ${deluge_SOURCES})
+target_sources(RunAllTests PRIVATE ${deluge_SOURCES})
```
[the remark this answers is not in the document]
Comment (stellar-aria) [reply tests/unit/CMakeLists.txt:34]: yep!

**Pull request merged**